ddtrace/tracer/option.go

```diff
@@ -813,6 +813,15 @@ func WithSpanID(id uint64) StartSpanOption {
 	}
 }
 
+// WithTraceID128High sets the higher-order 64 bits of a 128-bit trace id
+// of the started span if no Parent SpanContext is present,
+// overriding the random number that would be generated.
+func WithTraceID128High(id uint64) StartSpanOption {
+	return func(cfg *ddtrace.StartSpanConfig) {
+		cfg.TraceID128High = id
+	}
+}
```

Comment:
Is this a useful API for our users?

Do they ever want to set *just* the high bits, or should we have instead a `WithTraceID128(id string)` that sets both the low and high bits that they can use as a replacement for `WithTraceID`?

Reply:
I went back and forth on this a lot 🙃 

The reason I eventually opted for this is because of our current behavior for `WithSpanID`. That function will set the trace id to that id if it's a root span, and I was thinking about the interaction between those two functions. What do we do if someone provides an id to `WithSpanID` for a root span whose lower 64 bits are different from the lower 64 bits that are provided to `WithTraceID128`. Does `WithTraceID128` win in that case? That would be my inclination, and we could certainly document our way out of that situation, but I thought that would be a confusing UX.

I'm not honestly sure that anyone would ever need to use this, so we could just not have it at all until someone asks. But since we currently provide a way to set a trace id for root spans, I figured there were probably some customers who would also want to set the 128-bit trace id in that case too. I'm definitely open to thoughts here, since it's not straightforward.

Reply:
Another follow-up thought to this:
>  if someone provides an id to `WithSpanID` for a root span whose lower 64 bits are different from the lower 64 bits that are provided to `WithTraceID128`

If those lower bits don't match, would we also be expected to change the span id to match those? Otherwise the span id and trace id will be different for a root span, which will have subtle and incorrect implications.
Just in general, this feels like a recipe for confusion 😬